daemon/logger/loggerutils/logfile.go

```diff
@@ -371,7 +371,7 @@ func (w *LogFile) Close() error {
 	close(w.closed)
 	// Wait until any in-progress rotation is complete.
 	w.rotateMu.Lock()
-	w.rotateMu.Unlock() //nolint:staticcheck
+	w.rotateMu.Unlock() //nolint:staticcheck,gocritic
```

Comment:
We can do in a follow-up; looks like go-critic suggests using a `defer`, but haven't tried if that also satisfies `staticcheck`; https://go-critic.com/overview.html#badlock

So this could then be;

```go
w.rotateMu.Lock()
defer w.rotateMu.Unlock()
```

Or (from their example, making it a one-liner, perhaps to more clearly show intent, but not sure if we use that pattern elsewhere);

```go
w.rotateMu.Lock(); defer w.rotateMu.Unlock()
```